File: extract_testimonials.py
#!/usr/bin/env python3
"""
Extract all testimonials from Notion export with images and text
"""

import os
import json
import re
from pathlib import Path

BASE_PATH = r"C:\Users\marko yoga\Downloads\5fcd8bb5-1475-4692-a271-322995cb6cee_ExportBlock-00831fa7-3bc9-456e-8176-d071f91d1668\Private & Shared\Biblioteca de Depoimentos Mestre Ye"

def extract_metadata_from_md(md_content):
    """Extract metadata from markdown front matter"""
    metadata = {}
    lines = md_content.split('\n')

    for line in lines:
        # Extract title (first # heading)
        if line.startswith('# '):
            metadata['name'] = line.replace('# ', '').strip()

        # Extract key-value pairs
        if ':' in line and not line.startswith('#') and not line.startswith('!'):
            parts = line.split(':', 1)
            if len(parts) == 2:
                key = parts[0].strip()
                value = parts[1].strip()

                # Map Portuguese metadata keys
                key_mapping = {
                    'Formato': 'format',
                    'Tema Principal': 'main_theme',
                    'Canal': 'channel',
                    'Objeções Quebradas': 'objections_broken',
                    'Frase Principial': 'main_phrase',
                    'Idade': 'age'
                }

                mapped_key = key_mapping.get(key, key.lower().replace(' ', '_'))
                metadata[mapped_key] = value

    # Extract HEAD and DESTAQUE from markdown
    head_match = re.search(r'\*\*HEAD:?\*\*\s*(.+?)(?:\n|$)', md_content, re.IGNORECASE)
    if head_match:
        metadata['head'] = head_match.group(1).strip()

    destaque_match = re.search(r'\*\*DESTAQUE:?\*\*\s*["\']?(.+?)["\']?(?:\n|$)', md_content, re.IGNORECASE)
    if destaque_match:
        metadata['highlight'] = destaque_match.group(1).strip()

    return metadata

def find_images_for_folder(folder_path):
    """Find all images in a testimonial folder"""
    images = []
    folder = Path(folder_path)

    if folder.exists() and folder.is_dir():
        for ext in ['*.png', '*.jpg', '*.jpeg']:
            images.extend([str(img) for img in folder.glob(ext)])

    return images

def categorize_testimonial(metadata, content):
    """Categorize testimonial based on content"""
    categories = []
    tags = []

    # Combine all text for analysis
    full_text = ' '.join([
        metadata.get('name', ''),
        metadata.get('main_phrase', ''),
        metadata.get('head', ''),
        metadata.get('highlight', ''),
        metadata.get('objections_broken', ''),
        content
    ]).lower()

    # Health conditions
    if 'zumbido' in full_text:
        categories.append('ZUMBIDO')
        tags.append('zumbido')

    if 'labirintite' in full_text:
        categories.append('LABIRINTITE')
        tags.append('labirintite')

    if 'dor' in full_text or 'dores' in full_text:
        categories.append('DOR')
        if 'lombar' in full_text:
            tags.append('dor_lombar')
        if 'joelho' in full_text:
            tags.append('dor_joelho')
        if 'quadril' in full_text:
            tags.append('dor_quadril')
        if 'coluna' in full_text:
            tags.append('dor_coluna')
        if 'ombro' in full_text:
            tags.append('dor_ombro')

    if 'fibromialgia' in full_text:
        categories.append('FIBROMIALGIA')
        tags.append('fibromialgia')

    if 'ansiedade' in full_text:
        categories.append('ANSIEDADE')
        tags.append('ansiedade')

    if 'insônia' in full_text or 'insonia' in full_text or 'dormir' in full_text:
        categories.append('SONO')
        tags.append('insonia')

    if 'depressão' in full_text or 'depressao' in full_text:
        categories.append('DEPRESSAO')
        tags.append('depressao')

    if 'energia' in full_text or 'disposição' in full_text or 'disposicao' in full_text:
        tags.append('energia')
        tags.append('disposicao')

    # Extract percentages/numbers
    percentage_matches = re.findall(r'(\d+)%', full_text)
    if percentage_matches:
        tags.append(f"resultado_{max([int(p) for p in percentage_matches])}%")

    return list(set(categories)), list(set(tags))

def process_all_testimonials():
    """Process all testimonial markdown files"""
    testimonials = []
    base = Path(BASE_PATH)

    # Find all .md files
    md_files = list(base.glob('*.md'))

    print(f"Found {len(md_files)} markdown files to process...")

    for md_file in md_files:
        try:
            # Read markdown content
            with open(md_file, 'r', encoding='utf-8') as f:
                content = f.read()

            # Extract metadata
            metadata = extract_metadata_from_md(content)

            # Get folder name (remove hash suffix from filename)
            folder_name = md_file.stem
            # Remove hash suffix like "14910a87bb6b81d7bd44fb5e3aecff7d"
            folder_name = re.sub(r'\s+[a-f0-9]{32}$', '', folder_name)

            # Find corresponding folder and images
            folder_path = base / folder_name
            images = find_images_for_folder(folder_path)

            # Categorize
            categories, tags = categorize_testimonial(metadata, content)

            # Build testimonial object
            testimonial = {
                'id': md_file.stem,
                'client_name': metadata.get('name', folder_name),
                'testimonial_text': metadata.get('main_phrase', ''),
                'head': metadata.get('head', ''),
                'highlight': metadata.get('highlight', ''),
                'health_conditions': categories,
                'tags': tags,
                'age': metadata.get('age', ''),
                'channel': metadata.get('channel', ''),
                'format': metadata.get('format', ''),
                'main_theme': metadata.get('main_theme', ''),
                'objections_broken': metadata.get('objections_broken', ''),
                'images': [str(Path(img).name) for img in images],
                'image_paths': images,
                'folder_path': str(folder_path) if folder_path.exists() else '',
                'md_file': str(md_file)
            }

            testimonials.append(testimonial)

            # Print progress for important ones
            if any(cat in ['ZUMBIDO', 'LABIRINTITE', 'DOR'] for cat in categories):
                print(f"✓ {testimonial['client_name']}: {', '.join(categories)}")

        except Exception as e:
            print(f"Error processing {md_file.name}: {e}")

    return testimonials

def main():
    print("=" * 80)
    print("EXTRACTING ALL TESTIMONIALS FROM NOTION EXPORT")
    print("=" * 80)
    print()

    # Process all testimonials
    testimonials = process_all_testimonials()

    # Sort by categories (prioritize ZUMBIDO, LABIRINTITE, DOR)
    def sort_key(t):
        priority_categories = ['ZUMBIDO', 'LABIRINTITE', 'DOR', 'FIBROMIALGIA']
        for i, cat in enumerate(priority_categories):
            if cat in t['health_conditions']:
                return i
        return 999

    testimonials.sort(key=sort_key)

    # Statistics
    print()
    print("=" * 80)
    print("STATISTICS")
    print("=" * 80)
    print(f"Total testimonials: {len(testimonials)}")

    # Count by category
    category_counts = {}
    for t in testimonials:
        for cat in t['health_conditions']:
            category_counts[cat] = category_counts.get(cat, 0) + 1

    print("\nBy health condition:")
    for cat, count in sorted(category_counts.items(), key=lambda x: -x[1]):
        print(f"  {cat}: {count}")

    # Count with images
    with_images = sum(1 for t in testimonials if t['images'])
    print(f"\nTestimonials with images: {with_images}")

    # Save to JSON
    output_path = r"C:\projetos\quiz-mtc\testimonials-complete-with-images.json"
    with open(output_path, 'w', encoding='utf-8') as f:
        json.dump(testimonials, f, ensure_ascii=False, indent=2)

    print(f"\n✓ Saved to: {output_path}")
    print()

    # Show sample of top testimonials
    print("=" * 80)
    print("TOP TESTIMONIALS (ZUMBIDO/LABIRINTITE/DOR)")
    print("=" * 80)

    top = [t for t in testimonials if any(cat in ['ZUMBIDO', 'LABIRINTITE', 'DOR']
                                          for cat in t['health_conditions'])][:10]

    for i, t in enumerate(top, 1):
        print(f"\n{i}. {t['client_name']}")
        print(f"   Categories: {', '.join(t['health_conditions'])}")
        if t['testimonial_text']:
            print(f"   Text: {t['testimonial_text'][:100]}...")
        if t['images']:
            print(f"   Images: {len(t['images'])} files")

if __name__ == '__main__':
    main()

File: extract_testimonials.js
#!/usr/bin/env node
/**
 * Extract all testimonials from Notion export with images and text
 */

const fs = require('fs');
const path = require('path');

const BASE_PATH = String.raw`C:\Users\marko yoga\Downloads\5fcd8bb5-1475-4692-a271-322995cb6cee_ExportBlock-00831fa7-3bc9-456e-8176-d071f91d1668\Private & Shared\Biblioteca de Depoimentos Mestre Ye`;

function extractMetadataFromMd(mdContent) {
  const metadata = {};
  const lines = mdContent.split('\n');

  for (const line of lines) {
    // Extract title (first # heading)
    if (line.startsWith('# ')) {
      metadata.name = line.replace('# ', '').trim();
    }

    // Extract key-value pairs
    if (line.includes(':') && !line.startsWith('#') && !line.startsWith('!')) {
      const parts = line.split(':', 2);
      if (parts.length === 2) {
        const key = parts[0].trim();
        const value = parts[1].trim();

        // Map Portuguese metadata keys
        const keyMapping = {
          'Formato': 'format',
          'Tema Principal': 'main_theme',
          'Canal': 'channel',
          'Objeções Quebradas': 'objections_broken',
          'Frase Principial': 'main_phrase',
          'Idade': 'age'
        };

        const mappedKey = keyMapping[key] || key.toLowerCase().replace(/\s+/g, '_');
        metadata[mappedKey] = value;
      }
    }
  }

  // Extract HEAD and DESTAQUE from markdown
  const headMatch = mdContent.match(/\*\*HEAD:?\*\*\s*(.+?)(?:\n|$)/i);
  if (headMatch) {
    metadata.head = headMatch[1].trim();
  }

  const destaqueMatch = mdContent.match(/\*\*DESTAQUE:?\*\*\s*["\']?(.+?)["\']?(?:\n|$)/i);
  if (destaqueMatch) {
    metadata.highlight = destaqueMatch[1].trim();
  }

  return metadata;
}

function findImagesForFolder(folderPath) {
  const images = [];

  if (!fs.existsSync(folderPath) || !fs.statSync(folderPath).isDirectory()) {
    return images;
  }

  const files = fs.readdirSync(folderPath);
  for (const file of files) {
    const ext = path.extname(file).toLowerCase();
    if (['.png', '.jpg', '.jpeg'].includes(ext)) {
      images.push(path.join(folderPath, file));
    }
  }

  return images;
}

function categorizeTestimonial(metadata, content) {
  const categories = [];
  const tags = [];

  // Combine all text for analysis
  const fullText = [
    metadata.name || '',
    metadata.main_phrase || '',
    metadata.head || '',
    metadata.highlight || '',
    metadata.objections_broken || '',
    content
  ].join(' ').toLowerCase();

  // Health conditions
  if (fullText.includes('zumbido')) {
    categories.push('ZUMBIDO');
    tags.push('zumbido');
  }

  if (fullText.includes('labirintite')) {
    categories.push('LABIRINTITE');
    tags.push('labirintite');
  }

  if (fullText.includes('dor') || fullText.includes('dores')) {
    categories.push('DOR');
    if (fullText.includes('lombar')) tags.push('dor_lombar');
    if (fullText.includes('joelho')) tags.push('dor_joelho');
    if (fullText.includes('quadril')) tags.push('dor_quadril');
    if (fullText.includes('coluna')) tags.push('dor_coluna');
    if (fullText.includes('ombro')) tags.push('dor_ombro');
  }

  if (fullText.includes('fibromialgia')) {
    categories.push('FIBROMIALGIA');
    tags.push('fibromialgia');
  }

  if (fullText.includes('ansiedade')) {
    categories.push('ANSIEDADE');
    tags.push('ansiedade');
  }

  if (fullText.includes('insônia') || fullText.includes('insonia') || fullText.includes('dormir')) {
    categories.push('SONO');
    tags.push('insonia');
  }

  if (fullText.includes('depressão') || fullText.includes('depressao')) {
    categories.push('DEPRESSAO');
    tags.push('depressao');
  }

  if (fullText.includes('energia') || fullText.includes('disposição') || fullText.includes('disposicao')) {
    tags.push('energia');
    tags.push('disposicao');
  }

  // Extract percentages/numbers
  const percentageMatches = fullText.match(/(\d+)%/g);
  if (percentageMatches) {
    const percentages = percentageMatches.map(p => parseInt(p));
    const maxPercent = Math.max(...percentages);
    tags.push(`resultado_${maxPercent}%`);
  }

  return [Array.from(new Set(categories)), Array.from(new Set(tags))];
}

function processAllTestimonials() {
  const testimonials = [];

  // Find all .md files
  const files = fs.readdirSync(BASE_PATH);
  const mdFiles = files.filter(f => f.endsWith('.md'));

  console.log(`Found ${mdFiles.length} markdown files to process...`);

  for (const mdFile of mdFiles) {
    try {
      const mdPath = path.join(BASE_PATH, mdFile);

      // Read markdown content
      const content = fs.readFileSync(mdPath, 'utf-8');

      // Extract metadata
      const metadata = extractMetadataFromMd(content);

      // Get folder name (remove hash suffix from filename)
      const folderName = path.parse(mdFile).name.replace(/\s+[a-f0-9]{32}$/, '');

      // Find corresponding folder and images
      const folderPath = path.join(BASE_PATH, folderName);
      const images = findImagesForFolder(folderPath);

      // Categorize
      const [categories, tags] = categorizeTestimonial(metadata, content);

      // Build testimonial object
      const testimonial = {
        id: path.parse(mdFile).name,
        client_name: metadata.name || folderName,
        testimonial_text: metadata.main_phrase || '',
        head: metadata.head || '',
        highlight: metadata.highlight || '',
        health_conditions: categories,
        tags: tags,
        age: metadata.age || '',
        channel: metadata.channel || '',
        format: metadata.format || '',
        main_theme: metadata.main_theme || '',
        objections_broken: metadata.objections_broken || '',
        images: images.map(img => path.basename(img)),
        image_paths: images,
        folder_path: fs.existsSync(folderPath) ? folderPath : '',
        md_file: mdPath
      };

      testimonials.push(testimonial);

      // Print progress for important ones
      if (categories.some(cat => ['ZUMBIDO', 'LABIRINTITE', 'DOR'].includes(cat))) {
        console.log(`✓ ${testimonial.client_name}: ${categories.join(', ')}`);
      }
    } catch (e) {
      console.error(`Error processing ${mdFile}: ${e.message}`);
    }
  }

  return testimonials;
}

function main() {
  console.log('='.repeat(80));
  console.log('EXTRACTING ALL TESTIMONIALS FROM NOTION EXPORT');
  console.log('='.repeat(80));
  console.log('');

  // Process all testimonials
  const testimonials = processAllTestimonials();

  // Sort by categories (prioritize ZUMBIDO, LABIRINTITE, DOR)
  const priorityCategories = ['ZUMBIDO', 'LABIRINTITE', 'DOR', 'FIBROMIALGIA'];
  testimonials.sort((a, b) => {
    const getPriority = (t) => {
      for (let i = 0; i < priorityCategories.length; i++) {
        if (t.health_conditions.includes(priorityCategories[i])) {
          return i;
        }
      }
      return 999;
    };
    return getPriority(a) - getPriority(b);
  });

  // Statistics
  console.log('');
  console.log('='.repeat(80));
  console.log('STATISTICS');
  console.log('='.repeat(80));
  console.log(`Total testimonials: ${testimonials.length}`);

  // Count by category
  const categoryCounts = {};
  for (const t of testimonials) {
    for (const cat of t.health_conditions) {
      categoryCounts[cat] = (categoryCounts[cat] || 0) + 1;
    }
  }

  console.log('\nBy health condition:');
  const sortedCategories = Object.entries(categoryCounts).sort((a, b) => b[1] - a[1]);
  for (const [cat, count] of sortedCategories) {
    console.log(`  ${cat}: ${count}`);
  }

  // Count with images
  const withImages = testimonials.filter(t => t.images.length > 0).length;
  console.log(`\nTestimonials with images: ${withImages}`);

  // Save to JSON
  const outputPath = path.join('C:\\projetos\\quiz-mtc', 'testimonials-complete-with-images.json');
  fs.writeFileSync(outputPath, JSON.stringify(testimonials, null, 2), 'utf-8');

  console.log(`\n✓ Saved to: ${outputPath}`);
  console.log('');

  // Show sample of top testimonials
  console.log('='.repeat(80));
  console.log('TOP TESTIMONIALS (ZUMBIDO/LABIRINTITE/DOR)');
  console.log('='.repeat(80));

  const top = testimonials
    .filter(t => t.health_conditions.some(cat => ['ZUMBIDO', 'LABIRINTITE', 'DOR'].includes(cat)))
    .slice(0, 10);

  for (let i = 0; i < top.length; i++) {
    const t = top[i];
    console.log(`\n${i + 1}. ${t.client_name}`);
    console.log(`   Categories: ${t.health_conditions.join(', ')}`);
    if (t.testimonial_text) {
      console.log(`   Text: ${t.testimonial_text.substring(0, 100)}...`);
    }
    if (t.images.length > 0) {
      console.log(`   Images: ${t.images.length} files`);
    }
  }
}

main();
